docs/blog/2019-04-03-how-to-build-a-blog-with-wordpress-and-gatsby-part-1/index.md

```diff
@@ -0,0 +1,120 @@
+tags:
+  - wordpress
+  - apis
+  - blog
+  - headless cms
+  - react
+canonicalLink: https://www.iamtimsmith.com/blog/how-to-build-a-blog-with-wordpress-and-gatsby-part-1/
+---
+
+> Note: This is part one in a series of posts about Gatsby.js and WordPress. You can find [Part Two here](/blog/2019-04-04-how-to-build-a-blog-with-wordpress-and-gatsby-part-2), and [Part Three here](/blog/2019-04-05-how-to-build-a-blog-with-wordpress-and-gatsby-part-3).
+
+Ever wondered how to use Gatsby.js with WordPress? This post will cover everything you need to know about getting WordPress ready for a Gatsby site.
+
+In a [previous post](https://www.iamtimsmith.com/blog/using-wordpress-with-react/), I spoke about using React.js to build a frontend for a WordPress site. I talked about a simple setup to do this, but also mentioned some drawbacks to doing this.
+
+One such drawback is the difficulties faced when trying to rank for SEO, because search engine crawlers have a hard time indexing information being loaded into React. I mentioned briefly that there are ways around this and one of these ways is [Gatsby.js](https://www.gatsbyjs.org/).
+
+In this series of blog posts, I'll be talking about how to build a Gatsby.js site with a WordPress backend. The aim of the setup is to allow non-developer users to make changes on their website while allowing developers to use a fast, modern, and enjoyable tool to build with.
+
+<video autoplay muted loop>
+  <source src="https://media.giphy.com/media/3o7ZeTmU77UlPyeR2w/giphy.mp4">
+</video>
+
+## Why use Gatsby.js with WordPress at all?
+
+When I made my last post about using React.js with WordPress, a few people asked why one might do such a thing?
+
+There are a few reasons that creating a separate frontend in React.js is beneficial:
+
+First, it really allows a lot of customization for the developer to choose tools that will a) benefit the site most and b) create an enjoyable development experience. And let's be honest, as developers aren't we all looking for better development experience?
+
+Second, I believe that plugins are a double-edged sword. They allow some cool functionality which otherwise wouldn't exist for non-coders, but people get slap-happy with them and start putting them on unnecessarily. I've seen sites with more than 60 plugins activated at once and they wondered why their site was so slow! Having 60 different plugins could mean having a ridiculous amount of HTTP requests being sent because each plugin is loading a new library. Using a separate frontend means the developer is in control of this. It does mean that the client will have to pay to have some changes made, but at the same time it will ensure that the site is staying optimized and that a developer can make the change without adding a bunch of unnecessary junk.
+
+<video autoplay muted loop width="400">
```

Comment:
Can we add a caption here to describe what's in the video?

Reply:
@marcysutton Should I just add paragraph tags afterward or just add "alt text" inside of the video tags in the event that the video doesn't show up?

Reply:
I think the best approach would be to wrap the video in a `<figure>` element and use `<figcaption>`...it can be a short caption, but that way everyone can benefit from a little more context.

Reply:
I've added `<figure>` and `<figcaption>` tags to videos.